### ISSUE TITLE
Please add a new "File > Sync and Exit" menu command

### ISSUE THREAD

**Issue by unforgettableid** [Automattic/simplenote-electron#544]
Hi!  It would be good if you could please replace the "File > Exit" menu command with a new "File > Sync and Exit" command.

This command would do the following:

*  Show the word "Syncing..." onscreen.
*  Force a sync, including a check for new changes from the server.
*  Wait 1 second.
*  Show the word "Success" onscreen. And maybe a big green checkmark.
*  Wait 1 second.
*  Exit.

If the sync fails (e.g. due to a lack of Internet connectivity), Simplenote would show the following dialog box:

    Sync did not succeed.  Exit without syncing?
    
    [ Exit without syncing ]  [ Do&n't exit ]

Thoughts?

**Comment by dmsnell** [comment]
Thanks @unforgettableid. We have discussed the idea of indicating if there are changes in Simplenote which have not received positive confirmation of being received at the server.

On that _note_ we never actually trigger sync operations because it's always running as long as the app is open. There are ways to know for sure that changes we make locally have been received by the server but we cannot know anything more, such as being certain that changes weren't received by the server or whether we have the latest version of a note.

At best in specific circumstances we could say, "We definitely did not synchronize your changes because the network is unreachable. Would you like to wait and see if it comes back up?" But once it does come back we're left with uncertainty if the changes are definitely synchronized (though likely they will be).

@drw158 and I have chatted a bit about this and I'm tempted to close this specific issue because the challenge of knowing is much bigger than a specific menu item and the implications would probably change if we fixed the root issue. I'll keep it open for now and see if he has anything to add here.

**Comment by drw158** [comment]
Nothing to add. I think if we do anything, it would be a sync indicator as outlined in #459

**Comment by unforgettableid** [comment]
@dmsnell, in #459 you wrote:

> Since Simplenote is by nature a distributed application and updates to the same note could happen from different devices at near the same time we also run into race conditions with the updates, meaning that an announcement from the server could be on its way at the time we think we are up to date (meaning that we are now no longer up to date but don't realize it).

This is unlikely, but I admit that it's possible.  Especially in unusual cases like "my entire house's broadband Internet service just came back up after an outage".

OK, fine.  So you can't really guarantee that a "Sync and Exit" has been truly successful.  Instead of "Sync and Exit", perhaps you can instead provide a "Save and Exit" command.

A)  Can Simplenote know for sure that all local changes have been successfully received by the server?

B)  If so, could you make a "Save and Exit" command?  (This command would show the message "Sent data to cloud..." for five seconds.  Then, if all local changes have been successfully received by the server, it would show the words "Data saved successfully" for three seconds.  Then it would exit.)

**Comment by dmsnell** [comment]
@unforgettableid thanks for the followup. we actually do experience routine "unlikely" and "unusual" cases among the pool of Simplenote users. it's fun that way on account of the fact that these rare conditions do pop up and we are exposed to them, to try and make sure they don't cause any more trouble than necessary.

> provide a "Save and Exit" command.

it's not really about the specific words used. in the very moment before clicking on this hypothetical menu button, the app was already "saving."

we've definitely pointed our sights towards a small and limited indicator to try and better convey if we have a local queue of changes not yet sync'd.

> Can Simplenote know for sure that all local changes have been successfully received by the server?

herein lies the problem because the answer to this question is "no." _if_ it syncs successfully and reports back then yes, we can assert that. however, in the case where we don't hear back we can't.

> could you make a "Save and Exit" command?

going back to the discussion before, we need to figure out what this would mean. the successful path is safe: already done! the unsuccessful or _maybe_ unsuccessful path is the hard one. how long should we show a dialog saying, "we aren't sure if your changes have been synchronized; continue waiting?" we could probably assume three to five seconds but that even breaks down on a mobile network (for example) where we see packets occasionally come back successfully after twenty seconds or more (those long-lived responses are the worst!)

---

I'd like to understand better why it is that you feel like you need a save button. Did you experience data loss? Did the data not appear on another device when you expected it to? 

In many cases we might be fighting for the solution to a different problem, as one of Simplenote's primary goals is to make your notes instantly and reliably available on all your devices regardless of the quality of network connection (and yes, we apologize for the data loss issues we identified in some earlier versions of this app).

In other words, supposing that the notes were synchronized before you hit "save," would you still feel the need for the button?
